File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03862/p03862_num0_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    // Read input
    int N, x;
    cin >> N >> x;
    vector<int> a(N);
    for (int i = 0; i < N; ++i) {
        cin >> a[i];
    }
    
    // Process candies
    long long operations = 0;
    for (int i = 1; i < N; ++i) {
        int total = a[i] + a[i-1];
        if (total > x) {
            int diff = total - x;
            operations += diff;
            
            // Update neighboring boxes
            if (a[i] >= diff) {
                a[i] -= diff;
            } else {
                a[i] = 0;
                a[i-1] -= diff - a[i];
            }
        }
    }
    
    // Print result
    cout << operations << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03862/p03862_num0_parsed.py
import sys

# Read the input from stdin
n, x = map(int, sys.stdin.readline().split())
a = list(map(int, sys.stdin.readline().split()))

# Initialize the counter for number of operations required
operations = 0

# Iterate through the boxes
for i in range(1, n):
    # Find the excess candies in the neighboring boxes
    excess = max(0, a[i-1] + a[i] - x)
    
    # Update the number of operations required
    operations += excess
    
    # Eat the excess candies from the neighboring boxes
    a[i] = max(0, a[i] - excess)

# Print the minimum number of operations required
print(operations)